src/CommandObserver.ts

```diff
@@ -0,0 +1,96 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the Source EULA. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+'use strict';
+
+import * as vscode from 'vscode';
+import * as Constants from './constants';
+
+const regex = /^(,)?(((?<origin>([^\s].*))\((?<linedetails>(\d+|\d+,\d+|\d+,\d+,\d+,\d+))\))|()):\s(?<subcategory>(()|([^:]*? )))(?<category>(error|warning))(\s*(?<code>[^: ]*))?\s*:\s(?<text>.*)$/gm;
+
+export class CommandObserver {
+
+    private _outputChannel: vscode.OutputChannel = null;
+    private _diagnosticCollection: vscode.DiagnosticCollection = null;
+    private _diagnosticMap = new Map<string, vscode.Diagnostic[]>();
+
+    public next(message:string) {
+        this.setOutputChannel();
+        this._outputChannel.show(true);
```

Comment:
What does ```show``` do exactly? Should we move this into ```setOutputChannel```, so that it's only activated when created? If user selects some other output while it's building, I don't think we should constantly switch back when new message appears.

Reply:
This would bring Output window in focus. Like user could have Terminal pane open, but if we spit something out, this will be visible. Output channel is created only once when extension is loaded so that would not work. We should open this window each time user starts building, but yeah maybe not switch back and forth if they select some other pane while its building.

We can maybe give this responsibility to the caller. So the build command would make sure to show the  output channel when its starting.

Reply:
Either that or somehow switch on the first output... If there is nothing in the output - no reason to steal user's focus either. But for the first iteration I think switching there every time build is starting would be fine.